superset/connectors/connector_registry.py

```diff
@@ -44,12 +46,23 @@ def register_sources(cls, datasource_config: "OrderedDict[str, List[str]]") -> N
     def get_datasource(
         cls, datasource_type: str, datasource_id: int, session: Session
     ) -> "BaseDatasource":
-        return (
+        """Safely get a datasource without raising any errors.
+        Returns None if `datasource_type` is not registered or `datasource_id` does
+        not exists."""
```

Comment:
nit:
```suggestion
        not exist."""
```

Reply:
This comment is inaccurate. I decided to still raise an error to avoid making too many changes.

Will delete.